Non-Resx Embedded Resources get mangled names, possible name collisions

I was able to create a small repro 

the csproj looks like this
```csproj
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>netcoreapp3.0</TargetFramework>
  </PropertyGroup>
  <ItemGroup>
    <EmbeddedResource Include="MyResource.bmp" WithCulture="false" Type="Non-Resx" />
  </ItemGroup>
</Project>
```

Program.cs
```C#
namespace resourceName
{
    class Program
    {
        static void Main(string[] args)
        {
            Console.WriteLine(MyResource.Get()?.Length ?? -1);
        }
    }
}
```
MyResource.cs
```C#
namespace resourceName
{
    public static class MyResource
    {
        public static Stream Get() => typeof(MyResource).Assembly.GetManifestResourceStream($"{nameof(resourceName)}.{nameof(MyResource)}.bmp");
    }
}
```

Running this code under preview 7 gives output as 
```
C:\git\scratch\resourceName>dotnet.exe run
485514
```

where as running this code under preview 9 gives output as
```
C:\git\corefx>dotnet c:\git\scratch\resourceName\bin\Debug\netcoreapp3.0\resourceName.dll
-1
```


I checked dlls
the resource name include in .dll in previe9 is resourceName.MyResource where as under preview 7 is  resourceName.MyResource.bmp 


I hit this error while building corefx with preview9 for testData.resources file.

This works fine with 5.0 as well.

cc @ericstj @danmosemsft @livarcocc

Looks like there's another related failure mode here:

```
error CS1508: Resource identifier 'C' has already been used in this assembly"
```

From https://dev.azure.com/dnceng/public/_build/results?buildId=361408&view=results (thanks, @ViktorHofer).

We'll have to dig in, but I suspect this is the thing Nick predicted

> I imagine there will also be build errors if you have say Foo.txt and Foo.bmp as embedded resources next to Foo.cs. Without preserving the extension, this will cause duplicate resource names.

--- a/src/Tasks.UnitTests/CreateCSharpManifestResourceName_Tests.cs
+++ b/src/Tasks.UnitTests/CreateCSharpManifestResourceName_Tests.cs
@@ -417,6 +417,47 @@ namespace Microsoft.Build.UnitTests
         }
 
         /// <summary>
+        /// Opt into DependentUpon convention but don't expect it to be used for this file.
+        /// </summary>
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public void DependentUponConvention_DoesNotApplyToNonResx(bool explicitlySpecifyType)
+        {
+            using (var env = TestEnvironment.Create())
+            {
+                var csFile = env.CreateFile("SR1.cs", "namespace MyStuff.Namespace { class Class { } }");
+                const string ResourceFileName = "SR1.txt";
+                var resourceFile = env.CreateFile(ResourceFileName, "");
+
+                // Default resource naming is based on the item include, so use a relative
+                // path here instead of a full path.
+                env.SetCurrentDirectory(Path.GetDirectoryName(resourceFile.Path));
+                ITaskItem i = new TaskItem(ResourceFileName);
+                i.SetMetadata("BuildAction", "EmbeddedResource");
+                if (explicitlySpecifyType)
+                {
+                    i.SetMetadata("Type", "Non-Resx");
+                }
+                // Don't set DependentUpon so it goes by convention
+
+                CreateCSharpManifestResourceName t = new CreateCSharpManifestResourceName
+                {
+                    BuildEngine = new MockEngine(_testOutput),
+                    UseDependentUponConvention = true,
+                    ResourceFiles = new ITaskItem[] { i }
+                };
+
+                t.Execute().ShouldBeTrue("Expected the task to succeed.");
+
+                t.ManifestResourceNames.ShouldHaveSingleItem();
+
+                t.ManifestResourceNames[0].ItemSpec.ShouldBe(ResourceFileName, "Expecting to find the namespace & class name from SR1.cs");
+            }
+        }
+
+
+        /// <summary>
         /// Opt into DependentUpon convention and load the expected file properly when the file is in a subfolder.
         /// </summary>
         [Fact]

--- a/src/Tasks/CreateManifestResourceName.cs
+++ b/src/Tasks/CreateManifestResourceName.cs
@@ -146,8 +146,13 @@ namespace Microsoft.Build.Tasks
                     string fileName = resourceFile.ItemSpec;
                     string dependentUpon = resourceFile.GetMetadata(ItemMetadataNames.dependentUpon);
 
-                    // If opted into convention and no DependentUpon metadata, reference "<filename>.cs" if it exists.
-                    if (UseDependentUponConvention && string.IsNullOrEmpty(dependentUpon))
+                    string fileType = resourceFile.GetMetadata("Type");
+
+                    // If it has a file type metadata is "Resx" or has the file extension ".resx", we know it's a resx file.
+                    bool isResxFile = (!string.IsNullOrEmpty(fileType) && fileType == "Resx") || Path.GetExtension(fileName) == ".resx";
+
+                    // If opted into convention and no DependentUpon metadata and is a resx file, reference "<filename>.cs" if it exists.
+                    if (isResxFile && UseDependentUponConvention && string.IsNullOrEmpty(dependentUpon))
                     {
                         string conventionDependentUpon = Path.ChangeExtension(Path.GetFileName(fileName), SourceFileExtension);
 

--- a/src/Tasks/CreateManifestResourceName.cs
+++ b/src/Tasks/CreateManifestResourceName.cs
@@ -148,10 +148,11 @@ namespace Microsoft.Build.Tasks
 
                     string fileType = resourceFile.GetMetadata("Type");
 
-                    // If it has a file type metadata is "Resx"
+                    // If it has "type" metadata and the value is "Resx"
+                    // This value can be specified by the user, if not it will have been automatically assigned by the SplitResourcesByCulture target.
                     bool isResxFile = (!string.IsNullOrEmpty(fileType) && fileType == "Resx");
 
-                    //fall back onto the extension
+                    // If not, fall back onto the extension.
                     if (string.IsNullOrEmpty(fileType))
                     {
                         isResxFile = Path.GetExtension(fileName) == ".resx";

--- a/src/Tasks/CreateManifestResourceName.cs
+++ b/src/Tasks/CreateManifestResourceName.cs
@@ -148,8 +148,14 @@ namespace Microsoft.Build.Tasks
 
                     string fileType = resourceFile.GetMetadata("Type");
 
-                    // If it has a file type metadata is "Resx" or has the file extension ".resx", we know it's a resx file.
-                    bool isResxFile = (!string.IsNullOrEmpty(fileType) && fileType == "Resx") || Path.GetExtension(fileName) == ".resx";
+                    // If it has a file type metadata is "Resx"
+                    bool isResxFile = (!string.IsNullOrEmpty(fileType) && fileType == "Resx");
+
+                    //fall back onto the extension
+                    if (string.IsNullOrEmpty(fileType))
+                    {
+                        isResxFile = Path.GetExtension(fileName) == ".resx";
+                    }
 
                     // If opted into convention and no DependentUpon metadata and is a resx file, reference "<filename>.cs" if it exists.
                     if (isResxFile && UseDependentUponConvention && string.IsNullOrEmpty(dependentUpon))

